core/server/middleware/api/version-match.js

```diff
@@ -0,0 +1,16 @@
+var errors = require('../../errors');
+
+function checkVersionMatch(req, res, next) {
+    var requestVersion = req.get('X-Ghost-Version'),
+        currentVersion = res.locals.safeVersion;
+
+    if (requestVersion && requestVersion !== currentVersion) {
+        return next(new errors.VersionMismatchError(
+            'Request for version ' + requestVersion + ' does not match current version ' + currentVersion
```

Comment:
In https://github.com/TryGhost/Ghost/issues/6526 I outlined that I believe API errors should be translated at the point of display, rather than transmission. But I guess I should do it to keep consistency